api/controllers/v2/github.ts

```diff
@@ -87,6 +141,138 @@ export const TriggerBuild = async (event: APIGatewayEvent): Promise<APIGatewayPr
       body: errMsg,
     };
   }
+
+  let body: WorkflowRunCompletedEvent;
+  try {
+    body = JSON.parse(event.body) as WorkflowRunCompletedEvent;
+  } catch (e) {
+    console.log('[TriggerBuild]: ERROR! Could not parse event.body', e);
+    return {
+      statusCode: 502,
+      headers: { 'Content-Type': 'text/plain' },
+      body: ' ERROR! Could not parse event.body',
+    };
+  }
+
+  if (body.workflow_run.conclusion != 'success')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body:
+        'Build on branch ' +
+        body.workflow_run.head_branch +
+        ' is not complete and will not trigger smoke test site deployments ',
+    };
+
+  if (body.workflow_run.name != 'Deploy Staging ECS')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body:
+        'Workflow ' +
+        body.workflow_run.name +
+        ' completed successfully but only Deploy Staging ECS workflow completion will trigger smoke test site deployments',
+    };
+
+  // if the build was not building master, no need for smoke test sites
+  if (body.workflow_run.head_branch != 'main' || body.repository.fork) {
+    console.log('Build was not on master branch in main repo, sites will not deploy as no smoke tests are needed');
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body:
+        'Build on branch ' +
+        body.workflow_run.head_branch +
+        ' will not trigger site deployments as it was not on master branch in upstream repo',
+    };
+  }
+
+  //automated test builds will always deploy in dotcomstg
+  const env = 'dotcomstg';
+
+  async function createAndInsertJob() {
+    for (const s in SMOKETEST_SITES) {
+      const repoName = SMOKETEST_SITES[s];
+      const jobTitle = 'Smoke Test ' + repoName;
+      let repoInfo, projectEntry, repoOwner;
+
+      try {
+        repoInfo = await docsetsRepository.getRepo(repoName);
+        projectEntry = await projectsRepository.getProjectEntry(repoInfo.project);
+        repoOwner = projectEntry.github.organization;
+      } catch {
+        return 'repoInfo, projectEntry, or repoOwner not found for docs site ' + repoName;
+      }
+
+      const jobPrefix = repoInfo?.prefix ? repoInfo['prefix'][env] : '';
+
+      //add commit hash here: const newHead = body.workflow_run.head_sha;
```

Comment:
Just wondering what this is? A future ticket or something that can be erased?

Reply:
Future ticket! [DOP-4401](https://github.com/mongodb/docs-worker-pool/pull/1003). I can remove the comment now but it didn't seem like an issue to leave in since I'm already working on that future ticket